packages/core/data-transfer/lib/providers/local-strapi-destination-provider/index.ts

```diff
@@ -92,6 +96,49 @@ class LocalStrapiDestinationProvider implements IDestinationProvider {
     return mapSchemasValues(schemas);
   }
 
+  getAssetsStream(): NodeJS.WritableStream {
+    if (!this.strapi) {
+      throw new Error('Not able to stream Assets. Strapi instance not found');
+    }
+
+    const assetsDirectory = path.join(this.strapi.dirs.static.public, 'uploads');
+    const backupDirectory = path.join(
+      this.strapi.dirs.static.public,
+      `uploads_backup_${Date.now()}`
+    );
+
+    fse.renameSync(assetsDirectory, backupDirectory);
+    fse.mkdirSync(assetsDirectory);
+
+    return new Writable({
+      objectMode: true,
+      async final(next) {
+        fse.rmSync(backupDirectory, { recursive: true, force: true });
+        next();
+      },
+      async write(chunk, _encoding, callback) {
+        const entryPath = path.join(assetsDirectory, chunk.file);
+        const writableStream = fse.createWriteStream(entryPath);
+
+        chunk.stream
+          .pipe(writableStream)
+          .on('close', callback)
+          .on('error', (error: Error) => {
+            try {
+              fse.rmSync(assetsDirectory, { recursive: true, force: true });
+              fse.renameSync(backupDirectory, assetsDirectory);
```

Comment:
In general we should try to use the async methods with `await` over the *sync methods, because the sync functions are all blocking (node can't process anything else while it's waiting).

However, in these cases I don't think it matters a lot. If we've got an error there's not much being blocked, we're already aborting.

Reply:
Yes, that's why I didn't pay much attention to it, would you say it's worth changing it?